Source/Serilog.Exceptions/Reflection/ReflectionInfoExtractor.cs

```diff
@@ -0,0 +1,104 @@
+namespace Serilog.Exceptions.Reflection
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+
+    /// <summary>
+    /// Utility class that analyzes type using reflection and provides
+    /// information about properties to be used in destructuring process.
+    /// </summary>
+    internal class ReflectionInfoExtractor
+    {
+        private readonly object lockObj = new();
+        private readonly Dictionary<Type, ReflectionInfo> reflectionInfoCache = new();
+        private readonly IList<PropertyInfo> baseExceptionPropertiesForDestructuring;
+
+        public ReflectionInfoExtractor() => this.baseExceptionPropertiesForDestructuring = GetExceptionPropertiesForDestructuring(typeof(Exception));
+
```

Comment:
Very minor point. There are a few warnings that we can clean up.

Reply:
Fixed.